File: clsObj/linklst/linkedlist.java
public class linkedlist {

    private class Node {
        int data = 0;
        Node next = null;

        Node (int data) {
            this.data = data;
        }
    }

    private Node head = null;    
    private Node tail = null;
    private int size = 10;

    public int getSize() {
        return size;
    }

    public boolean isSizeEmpty() {
        return size == 0;
    }

    public void firstNode (int data) {
        Node node = new Node(data);
        addFirstNode(node);
    }
    public void addFirstNode(Node node) {
        if (isSizeEmpty()) {
            this.head = this.tail = node;
        }
        else {
            node.next = this.head;
            this.head = node;
        }
        this.size++;
    }

     public void lastNode(int data) {
        Node node = new Node(data);
        addLastNode(node);
    }

    private void addLastNode(Node node) {
        if (this.size == 0) {
            this.head = this.tail = node;
        } else {
            this.tail.next = node;
            this.tail = node;
        }
        this.size++;
    }

    public int removeFirst(int data) {
        if (this.size == 0) {
            return -1;
        }
        Node node = removeFirstNode();
        return node.data;
    }

    private Node removeFirstNode() {
        Node node = this.head;
        if (this.size == 1) {
            this.head = this.tail = null;
        } else {
            this.head = this.head.next;
            node.next = null;
        }
        this.size--;
        return node;
    }

     public int getFirst() {
        if (isSizeEmpty()) {
            return -1;
        }
        return getFirstNode().data;
    }

    private Node getFirstNode() {
        Node node = this.head;
        return node;
    }

    public int getLast() {
        if (this.size == 0) {
            return -1;
        }
        return getLastNode().data;
    }

    private Node getLastNode() {
        return this.tail;
    }

    public int getAt(int idx) {
        if (isSizeEmpty() || idx > this.size) {
            return -1;
        }
        Node node = getAtNode(idx);
        return node.data;
    }

    private Node getAtNode(int idx) {
        Node curr = this.head;
        while (idx-- > 0) {
            curr = curr.next;
        }
        return curr;
    }

        if (isEmpty()) {
            return -1;
        }

        Node node = removeLastNode();
        return node.data;
    }

    private Node removeLastNode() {
        Node node = this.tail;
        if (this.size == 1) {
            this.head = this.tail = null;
        } else {
            // for (int i = 0; i < size; i++) {
            // if (i == this.size-2) {
            // node.next = null;
            // this.tail = node;
            // } or
            // }
            Node secondLastNode = getAtNode(this.size - 2);
            secondLastNode.next = null;
            this.tail = secondLastNode;
        }
        this.size--;
        return node;
    }

}

File: clsObj/linklst/linkedlist.py
class LinkedList:

    class Node:
        def __init__(self, data):
            self.data = data
            self.next = None

    def __init__(self):
        self.head = None
        self.tail = None
        self.size = 10

    def get_size(self):
        return self.size

    def is_size_empty(self):
        return self.size == 0

    # -----------------------------------
    # Add First
    # -----------------------------------
    def first_node(self, data):
        node = self.Node(data)
        self.add_first_node(node)

    def add_first_node(self, node):
        if self.is_size_empty():
            self.head = self.tail = node
        else:
            node.next = self.head
            self.head = node

        self.size += 1

    # -----------------------------------
    # Add Last
    # -----------------------------------
    def last_node(self, data):
        node = self.Node(data)
        self.add_last_node(node)

    def add_last_node(self, node):
        if self.size == 0:
            self.head = self.tail = node
        else:
            self.tail.next = node
            self.tail = node

        self.size += 1

    # -----------------------------------
    # Remove First
    # -----------------------------------
    def remove_first(self, data):
        if self.size == 0:
            return -1

        node = self.remove_first_node()
        return node.data

    def remove_first_node(self):
        node = self.head

        if self.size == 1:
            self.head = self.tail = None
        else:
            self.head = self.head.next
            node.next = None

        self.size -= 1

        return node

    # -----------------------------------
    # Get First
    # -----------------------------------
    def get_first(self):
        if self.is_size_empty():
            return -1

        return self.get_first_node().data

    def get_first_node(self):
        return self.head

    # -----------------------------------
    # Get Last
    # -----------------------------------
    def get_last(self):
        if self.size == 0:
            return -1

        return self.get_last_node().data

    def get_last_node(self):
        return self.tail

    # -----------------------------------
    # Get At
    # -----------------------------------
    def get_at(self, idx):
        if self.is_size_empty() or idx > self.size:
            return -1

        node = self.get_at_node(idx)

        return node.data

    def get_at_node(self, idx):
        curr = self.head

        while idx > 0:
            curr = curr.next
            idx -= 1

        return curr

    def remove_last(self):
    if self.is_size_empty():
        return -1

    node = self.remove_last_node()
    return node.data


    def remove_last_node(self):
        node = self.tail
    
        if self.size == 1:
            self.head = self.tail = None
        else:
            second_last_node = self.get_at_node(self.size - 2)
    
            second_last_node.next = None
            self.tail = second_last_node
    
        self.size -= 1
    
        return node
